packages/blade/src/components/Button/_decisions/decisions.md

```diff
@@ -0,0 +1,81 @@
+# Button Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Internal BaseButton Component](#internal-basebutton-component)
+- [Button Component](#button-component)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=5200%3A0) to all variants of the Button component
+
+## Internal BaseButton Component
+-  Internal component that exposes certain extra props to enable creation of Component Level Action Buttons like `positive`, `negative`, `neutral`
+-  **BaseButton API**:
+   -  **`variant`**: `primary` | `secondary` | `tertiary` | `link`
+      -  **default**: `primary`
+      -  Link will look like an inline Link component but will be rendered as `button` element
+   -  **`intent`**: `positive` | `negative` | `notice` | `info` | `neutral`
+      -  **default**: `undefined`
```

Comment:
I feel the default value of a prop should only be one of the values allowed in the prop. Because it is unclear which `intent` would the Button be by default.

Reply:
That makes sense. If you see the Figma file, we're using intents only to tweak the variants further. Having an `undefined` intent makes it seem like something is missing. How about 
- `intent="none"`  (Seems awfully close to the existing `neutral`)
- `intent="action"` (I like this better since a default button is always used for some action but also confusing since positive, negative, notice, info, neutral are also used for an action)

Any suggestions?

Reply:
`none` sounds better to me as all kinds of buttons inherently perform some action. But I do agree it is somewhat close to `neutral`.

Reply:
@kamleshchandnani @anuraghazra thoughts?

Reply:
`intent="none"` sounds betters. Should it also be added as a type then?

Reply:
Works

Reply:
`intent: none` sounds like a new kind of intent altogether 😑. It's basically `undefined | one of positive, negative, neutral, info, notice`. we do it for all the properties because adding this at one place will force us introduce this pattern across of setting value of undefined props to be meaningful

Reply:
With most places we are sticking to the pattern of "default value should be one of the defined types" for instance, `isDisabled` is `false` by default whereas it could've been `undefined` by default and still worked. If we start considering `undefined` as a valid value instead of a "lack of value", we'd have to reconsider our defaults in places like `isDisabled` & `isFullWidth`

Although, `isDisabled={undefined}` will also be allowed as a valid value and will behave exactly like `isDisabled={false}`. By that logic, we should be considering `undefined` as a valid value and in this case having `intent={undefined}` should be valid and we could get rid of `intent="none"`

Reply:
I'll go with `undefined | possible unions` where undefined means it can be omitted and in that case the button will not have any intent specific styling as such.

Reply:
Leaning more toward having `undefined` as a valid value instead of `none`.

Reply:
Upvote this comment for `none`

Reply:
> I'll go with `undefined | possible unions` where undefined means it can be omitted and in that case the button will not have any intent specific styling as such.
> 
> Edit: also when we say `undefined | possible unions` the undefined key should not be explicitly the part of the union the whole prop should be optional. Otherwise it will force you to set `intent={undefined}` (Could be good or bad depending on if you want this or not.)

yep I love TS philosophy here: `intent?: one of FeedbackVariants` which basically says it's either undefined(doesn't exist) but if it exists then it has to be one of the variants value and internally in JS it's just the matter of if the prop exists then use that intent and apply respected styles else just ignore and none of the styles will be applied for intent

Reply:
I agree with `undefined` too after understanding the behaviour of `intent` and the analogy given by @kamleshchandnani.
`"none"` at this point is just us defining another form of `undefined`, but without the awesome TS, docs support that `undefined` comes with.